### PR TITLE
Add/update spec tests

### PR DESCRIPTION
**Pull request by Anoukh**
## Purpose
Add spec tests for error and table type

### CODE REVIEW

File: spec-conformance-tests/05-structured-values/tests/error.bal [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// reason-type-descriptor := type-descriptor
+// detail-type-descriptor := type-descriptor
+// The reason-type-descriptor must be a subtype of string; the
+// detail-type-descriptor must be a subtype of record { } (which is equivalent to
+// map<anydata|error>).
+@test:Config {}
+function testErrorTypeDescriptor() {
+    error error1 = error("Error One");
+    error<string> error2 = error("Error Two");
+    error<string, map<any>> error3 = error("Error Three", {detail: "failed"});
```
Comment (MaryamZi) [inline spec-conformance-tests/05-structured-values/tests/error.bal]: Isn't `map<any>` a deviation? Should be a sub-type of `map<anydata|error>` right?

Shall we also add a test with a map detail type descriptor, which also expects error?
```suggestion
    error<string, map<anydata|error>> error3 = error("Error Three", {detail: "failed", err: error("test error")});
```

Comment (MaryamZi) [reply spec-conformance-tests/05-structured-values/tests/error.bal]: Should we have a commented out test for stack trace also? Even something like 
```ballerina
   // test:assertNotEquals(error1.stackTrace(), ());
``` 
for the time-being?

File: spec-conformance-tests/05-structured-values/tests/deviations/error.bal [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+// TODO: Non-frozen values are accepted for the detail. Stack trace is not supported.
+// https://github.com/ballerina-platform/ballerina-lang/issues/13171
+@test:Config {
+    groups: ["broken"]
+}
+function testErrorTypeDescriptorBroken() {
+    map<anydata> detail = {cause: "Core Error"};
+    error<string, map<any>> error1 = error("Error Three", detail);
+    detail.stacktrace = "xyz";
+    test:assertEquals(error1.detail(), detail, msg = "expected detail map in error to be changed");
```
Comment (MaryamZi) [inline spec-conformance-tests/05-structured-values/tests/deviations/error.bal]: Should we have a commented out test for stack trace also? Even something like 
```ballerina
   // test:assertNotEquals(error1.stackTrace(), ());
``` 
for the time-being?

File: spec-conformance-tests/05-structured-values/tests/error.bal (hunk 1 above)
Comment (Anoukh) [reply spec-conformance-tests/05-structured-values/tests/error.bal]: Yes. Will fix

File: spec-conformance-tests/05-structured-values/tests/deviations/error.bal [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+function testErrorTypeDescriptorBroken() {
+    map<anydata> detail = { cause: "Core Error" };
+    error <string, map<any>> error1 = error("Error Three", detail);
+    detail.stacktrace = "xyz";
+    test:assertEquals(error1.detail(), detail, msg = "expected detail map in error to be changed");
+}
+
+// An error type does not have an implicit initial value.
+// TODO: Creating and error array should fail at compile time.
```
Comment (Shan1024) [inline spec-conformance-tests/05-structured-values/tests/deviations/error.bal]: Is there any issue created to track this?

**Pull request merged**